compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/declaration/FirOverrideChecker.kt

```diff
@@ -172,12 +173,30 @@ object FirOverrideChecker : FirClassChecker() {
         firTypeScope: FirTypeScope,
         context: CheckerContext,
     ) {
+        val overriddenMemberSymbols = firTypeScope.retrieveDirectOverriddenOf(member)
+
         if (!member.isOverride) {
+            if (overriddenMemberSymbols.isEmpty() ||
+                context.session.overridesBackwardCompatibilityHelper.overrideCanBeOmitted(overriddenMemberSymbols, context)
+            ) {
+                return
+            }
+            val kind = member.source?.kind
+            if (kind !is FirRealSourceElementKind && kind !is FirFakeSourceElementKind.PropertyFromParameter) return
```

Comment:
What is this check for? What are the overridden member fake kinds where override is not needed in overriding member? I was thinking it was for data class members but you can't subclass a data class. Is it property accessors? Please add a comment.

Reply:
It's for data class subclassing other classes and declare fields from primary constructor. Let me add a comment.

Reply:
I was just curious what false positives you'd get (if any) if you just removed this line altogether. (i.e., what other fake members can be overridden aside from property in primary constructor)

Reply:
I have seen synthetic members like `copy` in https://github.com/JetBrains/kotlin/blob/4c2cfd3ea981e911085959cdabc95bd8f3ecfa51/compiler/testData/diagnostics/tests/dataClasses/dataClassExplicitlyOverridingCopyWithDefaults.kt#L3-L2. Without this check, the class name is highlighted because the class body is the fake source for the synthetic copy method.